### DISCUSSION TITLE
Home Assistant Issue with node-red-contrib-ccu

### DISCUSSION THREAD

**Discussion by vdrecoll** [OpenCCU/OpenCCU#1703] [🇺🇸 General discussions]
I am using RaspberryMatic for years on an Asus Tinkerboard and I am very happy with the extraordinary support from @jens-maus  in maintaining the system. 

Now I am working on migrating the system to an RP4 running HomeAssistant OS and the Raspberry Addon. Since I am not programming on that level I am struggling a bit. I got it running and can access everything via the Addon UI. I can also embed the actuators on the HASS Dashboard and interact. 

Now I want to port my NodeRed flows. Howver I can not connect "node-red-contrib-ccu" to the Redmatic Addon. 

I am getting the message: "Error: rega getChannels connect ECONNREFUSED 192.168.178.101:8181"

In the addon I opened all ports. The IP address I am using to connect is the one from the HA RP4. In the installation WiKi from Jens  I am not getting section 7. in how to enable external access to the addon. Maybe that's it. Can anyone help with a few hints?

**Comment by Baxxy13** [comment]
Can you pls explain detailed where your NodeRed / Redmatic installation resides?

A: RedMatic inside the RM-HA-AddOn as an CCU/RM AddOn
B: NodeRed as an HA-AddOn (next to the RM-HA-AddOn)
C: NodeRed on any other host

**Reply by vdrecoll** [reply]
Hi Baxxy13, thanks for the answer. It's B NodeRed and Redmatic are installed as addon within the HA installation. 
In the installation WiKi under 7. there instructions to activate ports (see attached screenshot). However it doesn't accept the Host Id of the RM Addon (which works in the other configurations).

![2022-02-12 18_22_54-Home Assistant](https://user-images.githubusercontent.com/50049755/153721541-c895372d-0eea-4876-be72-adf611f665ed.gif)

**Reply by Baxxy13** [reply]
Luckily Option B is reatively easy to setup because all HA-AddOns reside in the same "HA-Network".

- There is no need to route any port out of the HA-RM-AddOn via the HA-RM-AddOn configuration.
- Set the firewall of the HA-RM-AddOn (WebUI) to fully open (just for testing, can be reduced later if needed *)
- Write down (or copy to a temporary txt-file) the hostname of the HA-RM-AddOn and the hostname of the HA-NodeRed-AddOn
(found in the respective AddOn information screen)
- Now in the HA-NodeRed-AddOn edit or setup the ccu-connection Node.
- Following Screenshot should be self explaining...

![HA_NodeRed_CCU_Connection_Setup](https://user-images.githubusercontent.com/54898249/153723431-0651569f-72d4-4464-adc6-dc873ded6f93.JPG)

A: Hostname of the HA-RM-AddOn
B: i'm not sure if this is really correct, but with this ip it works for me
C: Hostname of the HA-NodeRed-AddOn

It seems that after setting up, the ccu-connection needs some time to connect (or read the devices, variables etc.)
That's it. 


(* I leave the firewall settings (RM-WebUI) to fully open. I don't think that's a security problem because no ports are routed outside the "HA-Network".)

**Reply by vdrecoll** [reply]
Thanks a lot, looks like the answer I was looking for. Unfortunately during me trying things before I got your advice, I deleted the standard Homematic integration :-( I will set up the system from scratch again and post a feedback once it runs as intendet. 
However, appreciate your answers very much!

**Reply by PKR83** [reply]
Hi! Just stumbled over your post.
I am using the method you describes as A.
Same issue as vdrecoll mentioned.
What's the solution in that case?
